src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -2934,5 +2934,75 @@ public static void Main()
 ";
             CompileAndVerify(source, expected);
         }
+
+
+
+        [Fact]
+        [WorkItem(13734, "https://github.com/dotnet/roslyn/issues/13734")]
+        public void MethodGroupConversionNoSpill()
+        {
+            string source = @"
+using System.Threading.Tasks;
+using System;
+
+public class AsyncBug {
+    public static void Main() 
+    {
+        Boom().GetAwaiter().GetResult();
+    }
+    public static async Task Boom()
+    {
+        Func<Type> f = (await Task.FromResult(1)).GetType;
+        Console.WriteLine(f());
+    }
+}
+";
+
+            var v = CompileAndVerify(source, "System.Int32");
+        }
+        [Fact]
+
+        [WorkItem(13734, "https://github.com/dotnet/roslyn/issues/13734")]
+        public void MethodGroupConversionWithSpill()
+        {
+            string source = @"
+using System.Threading.Tasks;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+namespace AsyncBug
+{
+    class Program
+    {
+        private class SomeClass
+        {
+            public bool Method(int value)
+            {
+                return value % 2 == 0;
+            }
+        }
+
+        private Task<SomeClass> Danger()
+        {
```

Comment:
could you make this method async and add ```await Task.Yield()``` as a first statement - that is to make sure that callee gets suspended and to make sure spilling/re-hydrating is involved.

With just ```FromResult```  Task, it is very likely that the call returns synchronously.